### PR TITLE
node: prod vs dev, alternate directory stucture

### PR DESCRIPTION
**Pull request by sameersbn**
This is an alternate directory structure (wrt #105) for the prod 
and dev images and is currently support by the circle ci buildscripts.

In this structuring the `4/` directory consists of the Dockerfile for 
the production image, whereas the `4/dev` contains the Dockerfile
for the development image.

This directory structure and is currently being used to build the 
`bitnami/minideb-extras` flavors.

Merging this PR should build
 
 - `bitnami/node:4`
 - `bitnami/node:4.8.3-r1`
 - `bitnami/node:4-dev`
 - `bitnami/node:4.8.3-r1-dev`

~If we don't like the name `buildpack` for the development image, 
then it is a trivial change to support `bitnami/node:4-dev` or 
`bitnami/node:4-devel`~

edit: support for `dev` variant has been added to the circle ci build scripts via https://github.com/bitnami/test-infra/pull/35

### CODE REVIEW

**Comment by sameersbn** [comment]
For reference

```
REPOSITORY                         TAG                    IMAGE ID            CREATED             SIZE
bitnami/node                       4-buildpack            ae78d2663671        43 minutes ago      435MB
bitnami/node                       4                      b70d968c8bf8        About an hour ago   164MB
```

**Comment by sameersbn** [comment]
I wonder if we should use nami to unpack the node module as shown in the below snippet. This will install the nami installation metadata for `node` in `/root/.nami`. This would enable users to extend the image, install nami and have the node installation listed under `nami list`.

```dockerfile
RUN install_packages libssl1.0.0 wget ca-certificates && \
    wget -qc https://nami-prod.s3.amazonaws.com/tools/nami/releases/nami-0.0.6-0-linux-x64.tar.gz -P /tmp/ && \
    echo "08c474bf7b866879cd709f7c3efaebec43cfc2eb39c974fd83c18eb19821129a  /tmp/nami-0.0.6-0-linux-x64.tar.gz" | sha256sum -c - && \
    mkdir -p /opt/bitnami/nami && \
    tar xzf /tmp/nami-0.0.6-0-linux-x64.tar.gz --strip 1 -C /opt/bitnami/nami && \
    PATH=/opt/bitnami/nami/bin:$PATH && \
    wget -qc https://downloads.bitnami.com/files/stacksmith/node-8.1.3-0-linux-x64-debian-8.tar.gz -P /tmp/ && \
    echo "def9f116fe48414d959f161f290d4656fa4671594b1cd4c6e193220e4330b02d  /tmp/node-8.1.3-0-linux-x64-debian-8.tar.gz" | sha256sum -c - && \
    tar -zxf /tmp/node-8.1.3-0-linux-x64-debian-8.tar.gz -C /tmp/ && \
    nami install /tmp/node-8.1.3-0-linux-x64-debian-8 && \
    apt-get purge --auto-remove -y wget ca-certificates && \
    rm -rf /tmp/node-8.1.3-0-linux-x64-debian-8 /tmp/node-8.1.3-0-linux-x64-debian-8.tar.gz && \
    rm -rf /tmp/nami-0.0.6-0-linux-x64.tar.gz /opt/bitnami/nami
```

File: 4/dev/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,8 @@
+RUN install_packages build-essential git pkg-config unzip ghostscript imagemagick libbz2-1.0 libc6 libgcc1 libmysqlclient18 libncurses5 libreadline6 libsqlite3-0 libssl1.0.0 libstdc++6 libtinfo5 zlib1g
```
Comment (tompizmor) [inline 4/dev/Dockerfile]: why this Dockerfile does not have a FROM entry?

Comment (sameersbn) [inline 4/dev/Dockerfile]: This is how we've setup the ci build scripts. Basically the CI will build `4/Dockerfile` and then use the resulting image as the `FROM` for the `4/dev/Dockerfile`.

This is how it currently works with the CI script. We can however accommodate adding a `FROM` 
 line in `4/dev/Dockerfile` if we think it's required.

reference: https://github.com/bitnami/test-infra/blob/master/circle/docker-development-image.sh#L72-L80

**Comment by sameersbn** [comment]
> I would expect changes in the README file that explains the two variants per version
I will update the README to mention the two variants

>  and that the example includes a kubernetes.yaml manifest file

The PR updates the kubernetes.yml file to deploy the example app.  Do you want me to copy the kubernetes.yml file inside the `example/` directory?

File: 4/Dockerfile [hunk 2]
```diff
@@ -1,20 +1,17 @@
...
-MAINTAINER Bitnami <containers@bitnami.com>
+FROM bitnami/minideb:jessie
+LABEL maintainer "Bitnami <containers@bitnami.com>"
+
+RUN install_packages libssl1.0.0 wget ca-certificates && \
+    wget -qc https://downloads.bitnami.com/files/stacksmith/node-4.8.3-0-linux-x64-debian-8.tar.gz -P /tmp/ && \
+    echo "82ac604d6b97bd2070c8ff613de802096e5addd644d6817745ca2aa71d689aa3  /tmp/node-4.8.3-0-linux-x64-debian-8.tar.gz" | sha256sum -c - && \
+    mkdir -p /opt/bitnami && \
+    tar -zxf /tmp/node-4.8.3-0-linux-x64-debian-8.tar.gz -C /opt/bitnami/ --strip 2 node-4.8.3-0-linux-x64-debian-8/files && \
+    apt-get purge --auto-remove -y wget ca-certificates && \
```
Comment (anguslees) [inline 4/Dockerfile]: I presume we _want_ to leave libssl1.0.0 installed?

File: kubernetes.yml [hunk 3]
```diff
@@ -37,28 +45,52 @@ items:
...
+        - name: node-data
           persistentVolumeClaim:
-            claimName: .
+            claimName: node-data-pvc
   status: {}
 - apiVersion: v1
   kind: PersistentVolumeClaim
   metadata:
+    annotations:
+      volume.alpha.kubernetes.io/storage-class: default
```
Comment (anguslees) [inline kubernetes.yml]: time to drop this?  (now that 1.7 is out, and 1.6 introduced beta versions of these annotations)

File: 4/Dockerfile (hunk 2 above)
Comment (sameersbn) [inline 4/Dockerfile]: thats right. `python` which is part of the node nami module needs it.

File: kubernetes.yml (hunk 3 above)
Comment (sameersbn) [inline kubernetes.yml]: maybe you are right. but leaving the annotation would allow users to deploy the yaml on a pre 1.6 kubernetes cluster right?

**Comment by sameersbn** [comment]
closed by #107

**Pull request closed without merging**